lib/i18n/js/segment.rb

```diff
@@ -3,24 +3,25 @@ module JS
 
     # Class which enscapulates a translations hash and outputs a single JSON translation file
     class Segment
-      OPTIONS = [:namespace, :pretty_print, :js_extend, :sort_translation_keys].freeze
+      OPTIONS = [:namespace, :pretty_print, :js_extend, :sort_translation_keys, :locale_source].freeze
       LOCALE_INTERPOLATOR = /%\{locale\}/
 
       attr_reader *([:file, :translations] | OPTIONS)
 
       def initialize(file, translations, options = {})
-        @file         = file
-        @translations = translations
-        @namespace    = options[:namespace] || 'I18n'
-        @pretty_print = !!options[:pretty_print]
-        @js_extend    = options.key?(:js_extend) ? !!options[:js_extend] : true
+        @file          = file
+        @translations  = translations
+        @namespace     = options[:namespace] || 'I18n'
+        @pretty_print  = !!options[:pretty_print]
+        @js_extend     = options.key?(:js_extend) ? !!options[:js_extend] : true
         @sort_translation_keys = options.key?(:sort_translation_keys) ? !!options[:sort_translation_keys] : true
+        @locale_source = options[:locale_source] || "I18n.available_locales"
       end
 
       # Saves JSON file containing translations
       def save!
         if @file =~ LOCALE_INTERPOLATOR
-          I18n.available_locales.each do |locale|
+          eval(@locale_source).each do |locale|
```

Comment:
eval is evil, it can be used to execute arbitrary code. instead, define allowed formats for this option and implement a parsing function